src/date/format.js

```diff
@@ -207,6 +208,25 @@ return function( date, numberFormatters, properties ) {
 
 			// Zone
 			case "z":
+
+				// z...zzz: "{shortRegion}", eg. "PST" or "PDT".
+				// zzzz: "{regionName} {Standard Time}" or "{regionName} {Daylight Time}",
+				// eg. "Pacific Standard Time" or "Pacific Daylight Time".
+				if ( properties.tzId ) {
```

Comment:
Checking for `properties.tzId` isn't valid, because there's a case where `properties.tzId` is truthy and metaZone is unavailable and therefore the properties below are invalid (by looking at your code at date/format-properties.js). I think you should check for something else, e.g., one of the set properties.